src/components/Buttons/RadioCheckbox.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component, PropTypes }        from 'react'
+import Radium                                 from 'radium'
+import svgRadioBtnBase                        from '../../assets/radioBtnBase.svg'
+import svgRadioBtnChecked                     from '../../assets/radioBtnChecked.svg'
+import svgCheckboxBase                        from '../../assets/checkboxBase.svg'
+import svgCheckboxChecked                     from '../../assets/checkboxChecked.svg'
+
+const BASE_STYLES = {
+  input: {
+    margin: '0 10px -7px 0',
+    backgroundSize: 'contain',
+    backgroundRepeat: 'no-repeat',
+    backgroundPosition: 'top left',
+    backgroundColor: 'transparent',
+    appearance: 'none',
+    '-moz-appearance': 'none',
+    '-webkit-appearance': 'none',
+    width: 22,
+    height: 22,
+  },
+}
+
+const STYLES = {
+  radioBase: {
+    ...BASE_STYLES.input,
+    backgroundImage: `url(${svgRadioBtnBase.url})`,
+    ':active': {
+      backgroundImage: `url(${svgRadioBtnBase.url})`,
+      border: '0 solid transparent'
+    }
+  },
+  radioChecked: {
+    backgroundImage: `url(${svgRadioBtnChecked.url})`,
+    ':active': {
+      backgroundImage: `url(${svgRadioBtnChecked.url})`,
+      border: '0 solid transparent'
+    }
+  },
+  checkboxBase: {
+    ...BASE_STYLES.input,
+    backgroundImage: `url(${svgCheckboxBase.url})`,
+    ':active': {
+      backgroundImage: `url(${svgCheckboxBase.url})`,
+      border: '0 solid transparent'
+    }
+  },
+  checkboxChecked: {
+    backgroundImage: `url(${svgCheckboxChecked.url})`,
+    ':active': {
+      backgroundImage: `url(${svgCheckboxChecked.url})`,
+      border: '0 solid transparent'
+    }
+  }
+}
+
+class RadioCheckbox extends Component {
+  static propTypes = {
+    aria          : PropTypes.shape({
+      label         :PropTypes.string,
+    }),
+    btnType       : PropTypes.oneOf(['radio', 'checkbox']),
+    id            : PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
```

Comment:
We should auto generate an id if one is not provided. We do this in the in the `TextField` component, for example: https://github.com/instacart/Snacks/blob/master/src/components/Forms/TextField.js#L127

Reply:
Also, radio buttons, checkboxes and labels were one of the major problems found in our accessibility audit. That is the principal reason for this check.

Reply:
I would say we should stop requiring it, then.